core/src/channel/voice_buffer.rs

```diff
@@ -143,7 +143,9 @@ impl VoiceBuffer {
         }
 
         if let Some(max_voices) = max_voices {
-            if self.options.fade_out_killing {
+            if self.buffer.iter().filter(|v| v.id == id).count() > max_voices {
```

Comment:
Wait in what context is this condition even possible, what's your layer count? And how many voices do your SF2 spawners create per voice stack?

Reply:
You can debug this by using a soundfont that has 2 voices per key press (for example XP-80 fantasy) and set the layer count to 1. Before this change, two voices with the same ID would be added and ignored. So it would constantly try to reduce the voices to 1 while ignoring two voices and this would end up in an infinite loop, freezing the synth.
What I am skeptical about is the implementation for this fix because it might be slow when the buffers have a lot of voices. I'll think if I can come up with a better solution though.

Reply:
In my opinion, It's probably better to replace Iterator with ExactSizeIterator, and then do conditions based on that

Reply:
> In my opinion, It's probably better to replace Iterator with ExactSizeIterator (a rust builtin trait), and then do conditions based on that

I am guessing here you mean to use ExactSizeIterator for the voices parameter of the function. I tried this and it will need a lot more changes in the code so I added a counter of the added voices since we iterate over them anyway. If you meant something else let me know.